pkg/bitcoin/electrum/electrum.go

```diff
@@ -94,14 +93,21 @@ func (c *Connection) GetTransaction(
 			// See: https://github.com/Blockstream/electrs/pull/36
 			return client.GetRawTransaction(ctx, txID)
 		})
+	elapsed := time.Since(startTime)
 	if err != nil {
 		return nil, fmt.Errorf(
-			"failed to get raw transaction with ID [%s]: [%w]",
+			"failed to get raw transaction with ID [%s] after [%s]: [%w]",
```

Comment:
What do you think about doing it all directly in `requestWithRetry`? I think this can look as follows:
```
func requestWithRetry[K interface{}](
	c *Connection,
	requestFn func(ctx context.Context, client *electrum.Client) (K, error),
	requestName string,
) (K, error) {
	startTime := time.Now()
	logger.Infof("starting [%s] request to Electrum server", requestName)

	var result K

	err := wrappers.DoWithDefaultRetry(
		c.parentCtx,
		c.config.RequestRetryTimeout,
		func(ctx context.Context) error {
			if err := c.reconnectIfShutdown(); err != nil {
				return err
			}

			requestCtx, requestCancel := context.WithTimeout(ctx, c.config.RequestTimeout)
			defer requestCancel()

			c.clientMutex.Lock()
			r, err := requestFn(requestCtx, c.client)
			c.clientMutex.Unlock()

			if err != nil {
				return fmt.Errorf("request failed: [%w]", err)
			}

			result = r
			return nil
		})

	solveRequestOutcome := func(err error) string {
		if err != nil {
			return fmt.Sprintf("error: [%v]", err)
		}
		return "success"
	}
	
	logger.Infof("[%s] request to Electrum server completed with [%s] after [%s]",
		requestName,
		solveRequestOutcome(err),
		time.Since(startTime),
	)

	return result, err
}
```